tests/conftest.py

```diff
@@ -145,8 +145,68 @@ def global_execution_env():
             objective = client.add_objective(spec)
             objectives.append(objective)
 
-        assets = _TestAssets(datasets=datasets, objectives=objectives)
-        yield f, assets, n
+        assets = _DataEnv(datasets=datasets, objectives=objectives)
+        return assets
+
+
+@pytest.fixture
+def data_env_1(default_data_env, client_1):
+    """Fixture with pre-existing assets in first node."""
+    return default_data_env.filter_by(client_1.node_id)
+
+
+@pytest.fixture
+def data_env_2(default_data_env, client_2):
+    """Fixture with pre-existing assets in second node."""
+    return default_data_env.filter_by(client_2.node_id)
+
+
+@pytest.fixture
+def default_dataset(data_env_1):
+    """Fixture with pre-existing dataset in first node."""
+    return data_env_1.datasets[0]
+
+
+@pytest.fixture
+def default_objective(data_env_1):
+    """Fixture with pre-existing objective in first node."""
+    return data_env_1.objectives[0]
```

Comment:
We could indeed use `default_dataset_1` & `default_objective_1` here, but I think that it is easier to read this way when we only need assets from one client.

Reply:
👍